pkg/api/v1alpha1/etcdencryption_types.go

```diff
@@ -0,0 +1,21 @@
+package v1alpha1
+
+// EtcdEncryption defines the configuration for ETCD encryption.
+type EtcdEncryption struct {
+	Providers []EtcdEncryptionProvider `json:"providers"`
+	// Resources defines a list of objects and custom resources definitions that should be encrypted.
+	Resources []string `json:"resources"`
```

Comment:
question about this bc I don't know how it works
would all resources we encrypted with all providers?

Reply:
You can have something like this where you configure one provider for some resources and another provider for other resources. 

```yaml
spec:
  etcdEncryption:
  - providers:
    - kms:
        socketListenAddress: unix:///var/run/kmsplugin/socket1-new.sock
    - kms:
        socketListenAddress: unix:///var/run/kmsplugin/socket1-old.sock
    resources:
    - secrets
    - custom-resource-1.anywhere.eks.amazonsaws.com
  - providers:
    - kms:
        socketListenAddress: unix:///var/run/kmsplugin/socket2-new.sock
    - kms:
        socketListenAddress: unix:///var/run/kmsplugin/socket2-old.sock
    resources:
    - configmaps
    - custom-resource-2.anywhere.eks.amazonsaws.com
```

In this example, `providers[0]` will be used to encrypt `secrets` and `custom-resource-1.anywhere.eks.amazonsaws.com` while `providers[1]` will be used to `configmaps` and `custom-resource-2.anywhere.eks.amazonsaws.com`